documentation/feature-flags.md

```diff
@@ -0,0 +1,51 @@
+# Node Agent Feature Flags
+
+As New Relic develops new instrumentation features, we may release some of them under feature flags--disabling them unless the agent is started with an optional flag to enable them. We do this to avoid side-effects on instrumented code when these still-experimental features are in development.  
```

Comment:
Overall this looks great. A few thoughts:

 * Have more structure explaining the feature flags

```
#### <feature_flag_name>
configuration: `{ feature_flag : { <feature_flag_name>: true|false }}`
environ variable: `NEW_RELIC_FEATURE_FLAG_<feature_flag_name>`
description: lorem ipsum...
```